lib/database/leaderboard.php

```diff
@@ -460,13 +479,16 @@ function GetLeaderboardData($lbID, $user, $numToFetch, $offset, $friendsOnly)
                 $numResultsFound++;
             }
 
-            if ($userFound == false) {
+            if ($userFound == false && $nearby == 0) {
```

Comment:
i don't believe it's necessary to append the user entry when not doing a `nearby` query. the caller either wants N records around the user, or N records at some offset.

i realize that this would be a breaking change, but if the user is known, this should be a nearby query and this code won't be getting called anyway.

you could potentially eliminate all the code related to tracking whether or not the user was found, the query, and appending the record.

Reply:
GetLeaderboardData() is also used for the website leaderboards and this part is for appending the user to the end of the list there which is why I made some of the decisions around this such as the nearby variable. At this point though it may be better to split the lbinfo call to its own unique version of GetLeaderboardData()

Reply:
good point about the other usage.in that case, i think it's fine the way you have it.